components/AIEstimation.tsx

```diff
@@ -0,0 +1,60 @@
+// components/ai-estimation.tsx
+'use client'
+
+import { useState } from 'react'
+import { useMutation } from 'convex/react'
+import { api } from '@/convex/_generated/api'
+import { Button } from "@/components/ui/button"
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
+import { Spinner } from "@/components/ui/spinner"
+import { useToast } from '@/hooks/use-toast'
+
+interface AIEstimationProps {
+    vehicleDetails: any // Replace with proper type
+    selectedServices: string[]
+    customizations: any // Replace with proper type
+}
+
+export function AIEstimation({ vehicleDetails, selectedServices, customizations }: AIEstimationProps) {
+    const [estimationResult, setEstimationResult] = useState<{ estimatedTotal: number, detailedAnalysis: string } | null>(null)
+    const calculateEstimate = useMutation(api.estimations.calculate)

```

Comment:
**Suggestion:** Consider adding error handling for the case when `calculateEstimate` mutation fails, to prevent the application from crashing and to provide a better user experience. [error handling, importance: 8]
```suggestion
    const calculateEstimate = useMutation(api.estimations.calculate, {
        onError: (error) => {
            console.error("Failed to calculate estimate:", error);
            useToast({
                title: "Error",
                description: "Failed to calculate estimate due to server error",
                variant: "destructive"
            });
        }
    })
```